src/openrct2/object/ObjectRepository.cpp

```diff
@@ -525,7 +540,7 @@ class ObjectRepository final : public IObjectRepository
         // Encode data
         ObjectType objectType = entry->GetType();
         SawyerCodingChunkHeader chunkHeader;
-        chunkHeader.encoding = object_entry_group_encoding[EnumValue(objectType)];
+        chunkHeader.encoding = kLegacyObjectEntryGroupEncoding[EnumValue(objectType)];
```

Comment:
This function is only used to save legacy objects. However, this is not guarded anywhere. Accessing `object_entry_group_encoding` using objectType as an index (max 18) could lead to out of bounds, as the array only has definitions for the 11 vanilla object types. I haven't changed this, but at least `std::array` has bounds checking.